### PR TITLE
Fixed a bug with ODATA update/create of entities

### PR DESCRIPTION
**Pull request by mit-ko**
Fixed an issue creating wrong sql for post/put/merge of OData2 entities

Added OData2 tests for post/put/merge

Added compatibility for java 1.8 for the OData2 core module

### What does this PR do?

Fixed a bug that prevents post/put/merge of OData2 entities

### CODE REVIEW

**Comment by mit-ko** [comment]
Would it be also possible to merge it in the latest 5.x branch?
Thanks in advance

File: modules/odata/odata-core/pom.xml [hunk 1]
```diff
@@ -21,12 +21,20 @@
 				<artifactId>maven-compiler-plugin</artifactId>
 				<version>${maven.compiler.plugin.version}</version>
 				<configuration>
-					<source>${maven.compiler.source}</source>
-					<target>${maven.compiler.target}</target>
+					<source>1.8</source>
```
Comment (delchev) [inline modules/odata/odata-core/pom.xml:24]: Do we need to set it explicitly to 1.8? Currently it is 11 which supports the features used below

File: modules/odata/odata-core/pom.xml [hunk 2]
```diff
@@ -21,12 +21,20 @@
...
+					<target>1.8</target>
```
Comment (delchev) [inline modules/odata/odata-core/pom.xml:25]: Same here - it is 11 already

**Review by delchev — CHANGES_REQUESTED** [review]

File: modules/odata/odata-core/pom.xml (hunk 1 above)
Comment (mit-ko) [reply modules/odata/odata-core/pom.xml:24]: The thing is that when a module is compiled with java 11 it cannot be used with projects that run on java 1.8

File: modules/odata/odata-core/pom.xml (hunk 2 above)
Comment (mit-ko) [reply modules/odata/odata-core/pom.xml:25]: Same reason here

**Review by ThuF — APPROVED** [review]
+1

**Review by delchev — APPROVED** [review]

**Pull request merged**